tracer/src/Datadog.Trace/AspNet/TracingHttpModule.cs

```diff
@@ -279,6 +279,11 @@ private void OnEndRequest(object sender, EventArgs eventArgs)
 
                         if (Iast.Iast.Instance.Settings.Enabled && IastModule.AddRequestVulnerabilitiesAllowed())
                         {
+                            if (rootSpan is not null && HttpRuntime.UsingIntegratedPipeline && _canReadHttpResponseHeaders)
+                            {
+                                ReturnedHeadersAnalyzer.Analyze(app.Context.Response.Headers, IntegrationId, rootSpan.ServiceName, app.Context.Response.StatusCode);
```

Comment:
So there's a small potential issue here, ironically _because_ the `ReturnedHeadersAnalyzer.Analyze()` method has a global `catch` clause. As the comments above mention:

```
// Despite the HttpRuntime.UsingIntegratedPipeline check, we can still fail to access response headers, for example when using Sitefinity: "This operation requires IIS integrated pipeline mode"
```

With the way it's currently written, if we hit that issue here, _either_ this will throw _before_ calling `Analyze()` _or_ it will throw a `PlatformNotSupportedException` _inside_ `Analyze()`, which will be caught, but means that we _won't_ set `_canReadHttpResponseHeaders` here, which could cause potential perf issues.

There are various solutions to this (e.g. add an exception filter to `Analyze()`, but I think the simplest is to remove the try-catch from inside `Analyze()`, and instead require callers to catch the exception. e.g. 


I think you can work aounneed to wrap this in a try catch (see the comment earlier in the file
```suggestion
                                try
                                {
                                    ReturnedHeadersAnalyzer.Analyze(app.Context.Response.Headers, IntegrationId, rootSpan.ServiceName, app.Context.Response.StatusCode);
                                }
                                catch (PlatformNotSupportedException ex)
                                {
                                    // Despite the HttpRuntime.UsingIntegratedPipeline check, we can still fail to access response headers, for example when using Sitefinity: "This operation requires IIS integrated pipeline mode"
                                    Log.Error(ex, "Unable to access response headers when analyzing headers. Disabling for the rest of the application lifetime.");
                                    _canReadHttpResponseHeaders = false;
                                }
                                catch (Exception ex)
                                {
                                    Log.Error(ex, "Error analyzing HTTP response headers");
                                }
```

Reply:
That's a good one. Thanks!